### PR TITLE
auro-radio: update label to use slot

### PR DESCRIPTION
**Pull request by blackfalcon**
# Alaska Airlines Pull Request

**Fixes:** #40 

## Summary:

The scope of this update is to extend the placement of label content to use a pre-defined slot as well as add a new `content` slot that allows for greater creative content use within the scope of auro-radio. 

This is **NOT** a breaking change in the shape of the element. Users will still be allowed to add content via the `label` attribute, but using the slot will override the string from the attribute as shown in the following screenshot.

The display of the markup in the `content` slot is **NOT** managed by the auro-radio custom element. This markup is in the `light DOM` and able to be styled by the consumer.  

Screen Shot 2021-07-03 at 1 11 07 PM

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -68,6 +68,27 @@ <h1>auro-radio</h1>
         </template>
       </demo-snippet>
 
+      <p>Group w/slots</p>
+      <demo-snippet>
+        <template>
+            <auro-radio-group required>
+              <span slot="legend">Will you pick a plan?</span>
+              <auro-radio id="radio10" name="radioDemo10" value="yes">
+                <b>Lounge $300 USD</b><br>
```
Comment (jason-capsule42) [inline demo/index.html:77]: The break here isn't necessary (I tested the rendering with it removed) for the content to appear below the label. Including it lead me to initially think it was required to have the content slot not appear inline with the label.

```suggestion
                <b>Lounge $300 USD</b>
```

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
The line break on line 77 of the index.html (see comment in changes) should be removed.

**Comment by Dayna-A** [comment]
Hey @blackfalcon tabbing through the demo page seems to jump from the top option in one auro-radio group to the middle option in the next to a different one in the one after that? Is this a side effect of the way the demo page is set up or something or is there cause for concern?

**Comment by Dayna-A** [comment]
Is the tab focus supposed to jump to disabled radio groups? in the demo page in chrome on mac it seems to skip the individually disabled ones but not the disabled group for me.. screen recording: https://drive.google.com/file/d/13eODKqjOAX1c2_UctUdCPOZGFXkx-wjC/view

**Review by Dayna-A — DISMISSED** [review]

**Comment by blackfalcon** [comment]
@Dayna-A nope, that is a new bug. See https://github.com/AlaskaAirlines/auro-radio/issues/48

This new bug was NOT caused by this PR so I do not want to conflate this PR to address the bug.

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**